Description: If the Ogg or FLAC filetypes have been flagged as being stored as
 "unknown" and the device now indicates it supports them then remove the flag.
Origin: upstream, commit:1147
---
 src/libmtp.c |   25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

--- libmtp.orig/src/libmtp.c
+++ libmtp/src/libmtp.c
@@ -1798,6 +1798,31 @@ LIBMTP_mtpdevice_t *LIBMTP_Open_Raw_Devi
 		 "Trying to continue anyway.");
   }
 
+  /*
+   * If the OGG or FLAC filetypes are flagged as "unknown", check
+   * if the firmware has been updated to actually support it.
+   */
+  if (FLAG_OGG_IS_UNKNOWN(ptp_usb)) {
+    for (i=0;i<current_params->deviceinfo.ImageFormats_len;i++) {
+      if (current_params->deviceinfo.ImageFormats[i] == PTP_OFC_MTP_OGG) {
+        /* This is not unknown anymore, unflag it */
+        ptp_usb->rawdevice.device_entry.device_flags &=
+          ~DEVICE_FLAG_OGG_IS_UNKNOWN;
+        break;
+      }
+    }
+  }
+  if (FLAG_FLAC_IS_UNKNOWN(ptp_usb)) {
+    for (i=0;i<current_params->deviceinfo.ImageFormats_len;i++) {
+      if (current_params->deviceinfo.ImageFormats[i] == PTP_OFC_MTP_FLAC) {
+        /* This is not unknown anymore, unflag it */
+        ptp_usb->rawdevice.device_entry.device_flags &=
+          ~DEVICE_FLAG_FLAC_IS_UNKNOWN;
+        break;
+      }
+    }
+  }
+
   /* Determine if the object size supported is 32 or 64 bit wide */
   for (i=0;i<current_params->deviceinfo.ImageFormats_len;i++) {
     PTPObjectPropDesc opd;
